viewer/packages/cad-styling/src/TreeIndexNodeCollection.ts

```diff
@@ -43,14 +49,51 @@ export class TreeIndexNodeCollection extends NodeCollectionBase {
     return this._treeIndices;
   }
 
+  getAreas(): AreaCollection {
+    if (!this._areaCollection) {
+      if (this._treeIndices.count === 0) {
+        return EmptyAreaCollection.instance();
+      }
+
+      throw new Error(
+        `The AreaCollection returned by getAreas() for ThreeIndexNodeCollection must be constructed manually using addAreas() and addAreaPoints()`
+      );
+    }
+
+    return this._areaCollection;
+  }
+
+  addAreas(areas: THREE.Box3[]): void {
+    if (!this._areaCollection) {
+      this._areaCollection = new ClusteredAreaCollection();
+    }
+
+    this._areaCollection.addAreas(areas);
+  }
+
+  addAreaPoints(points: THREE.Vector3[]): void {
```

Comment:
It should be Tsdoc'ed what this actually does

Reply:
How does it look now?